### PR TITLE
[Enhencement] Add glowroot pugins

### PR DESCRIPTION
**Pull request by chibenwa**
Added pluging:
 - SMTP: Allow review of incoming SMTP traffic, with a per-command analysis
 - JMAP: Allow review per method

Because so far JMAP relies on HTTP plugin, all method calls are presented under the same web transaction
which prevent us from doing a in depth per method review.

This new JMAP plugin aims at enhancing this.

 - MailboxListener: monitor per mailbox listener consumed resources
 - Task: Allow a per task performance analysis

Enhenced plugins:
 - Spooler: add recording per Mailet / Matcher to contextualize more spooler statistics

For some products, instrumenbtation for unsupported backends/protocols can be removed

### CODE REVIEW

File: dockerfiles/run/guice/cassandra-ldap/destination/glowroot/plugins/mailboxListener.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "MailboxListener Plugin",
+  "id": "jmap",
```
Comment (trantienduchn) [inline dockerfiles/run/guice/cassandra-ldap/destination/glowroot/plugins/mailboxListener.json:3]: should it be something else?

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/glowroot/plugins/task.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "JMAP Plugin",
```
Comment (trantienduchn) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/glowroot/plugins/task.json:2]: idem

**Comment by mbaechler** [comment]
Thank you for the description of this PR. Could you please show us with some screenshots what we'll get?

**Comment by chibenwa** [comment]
Capture d’écran de 2020-04-06 19-02-53

Capture d’écran de 2020-04-06 19-04-43

Capture d’écran de 2020-04-06 19-09-38

Capture d’écran de 2020-04-06 19-15-49


Here you are!

Looks like the changes I proposed for Mailet/matcher & JMAP to not have effect (if anyone has an idea???)

**Comment by chibenwa** [comment]
Merged

**Comment by rouazana** [comment]
I don't like the [Enhencement] prefix (even with a typo in it ;)). I'm ok with the [Refactor] typo as long as it means no change in the functionality. But for every other case I would prefer than you open a JIRA.

**Pull request closed without merging**